### PR TITLE
add data export data comparison script

### LINKED ISSUE #1512: Data Export: Data comparison
- [x] write initial script to compare data and produce differences 
- [x] verify diffs and note issues #1533

### LINKED ISSUE #1533: Data migration fixes
- [x] growingStockComposition.remaining_introduced_placeholder

### PR DESCRIPTION
**Pull request by minotogna**
- added migrate-fra:data-export:compare yarn command 
- added ts code under src/test/dataExportComparison
- added src/test/sectionSpec legacy metadata. Note: it has been added just to use legacy spec types, ignore any inconsistencies.
- moved fraSpecs.ts from tools/dataMigration to src/test/sectionSpec/
- fixed data migration issues found from diffs result 

resolves #1512
resolves #1533

### CODE REVIEW

File: src/test/dataExportComparison/getMetadata.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    metadata: MetadataLocal
+  }>(`
+      with d as (select t.props ->> 'name'                            as name,
+                        t.props -> 'columnsExport' -> '${cycle.uuid}' as columns,
+                        jsonb_agg(r.props ->> 'variableName')         as variables
+                 from ${schema}."table" t
+                          left join ${schema}.table_section ts on ts.id = t.table_section_id
+                          left join ${schema}.section s on s.id = ts.section_id
+                          left join ${schema}.row r on t.id = r.table_id
+                 where (s.props -> 'dataExport')::boolean
```
Comment (rfontanarosa) [inline src/test/dataExportComparison/getMetadata.ts]: where (s.props ->> 'dataExport')::boolean
                       and (t.props ->> 'dataExport')::boolean
                       and (r.props -> 'variableName') is not null

**Review by sorja — APPROVED** [review]

**Review by rfontanarosa — COMMENTED** [review]
diff.csv found a compare error with AUS (extentOfForest - otherLand)

in production those values exists in the country section page but not in data export

Schermata 2022-09-21 alle 11 26 32

Schermata 2022-09-21 alle 11 34 02

**Comment by minotogna** [comment]
That's why it's a bug in prod 
That means our local version doesn't have the bug therefore there are no errors in data comparison 
Therefore this is not an issue

**Comment by rfontanarosa** [comment]
> at means our local version doesn't have the bug therefore there are no errors in data comparison
> Therefore this is not an issue

great! so, these is only a query to fix and it's ok for us

**Comment by minotogna** [comment]
There's nothing to fix from our side

**Review by rfontanarosa — APPROVED** [review]

**Pull request merged**